File: 26-remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int r = 0;
        int l = 0;
        int n = nums.length;
        while(r < n){
            nums[l] = nums[r];
            while(r < n && nums[l] == nums[r]){
                r ++;
            }
            l++;
        }
        return l;
    }
}

File: 26-remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
      int l=0;
      int r=0;
      int n=nums.size();
      while(r<n){
        nums[l]=nums[r];
        while(r<n&& nums[l]==nums[r]){
            r++;
        }
        l++;
      }
      return l;
    }
};